src/Integration.Vsix/CFamily/FileConfigProvider.cs

```diff
@@ -0,0 +1,116 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using EnvDTE;
+using SonarLint.VisualStudio.Core.CFamily;
+using SonarLint.VisualStudio.Integration.Helpers;
+
+namespace SonarLint.VisualStudio.Integration.Vsix.CFamily
+{
+    internal interface IFileConfigProvider
+    {
+        CFamilyHelper.FileConfig Get(ProjectItem projectItem, string analyzedFilePath, CFamilyAnalyzerOptions analyzerOptions);
+    }
+
+    internal class FileConfigProvider : IFileConfigProvider
+    {
+        private static readonly NoOpLogger noOpLogger = new NoOpLogger();
+        private readonly ILogger logger;
+
+        public FileConfigProvider(ILogger logger)
+        {
+            this.logger = logger;
+        }
+
+
+        public CFamilyHelper.FileConfig Get(ProjectItem projectItem, string analyzedFilePath, CFamilyAnalyzerOptions analyzerOptions)
+        {
+            var analysisLogger = GetAnalysisLogger(analyzerOptions);
+
+            if (!IsFileInSolution(projectItem))
+            {
+                analysisLogger.LogDebug($"[CFamilyHelper] Unable to retrieve the configuration for file '{analyzedFilePath}'. Check the file is part of a project in the current solution.", analyzerOptions);
```

Comment:
Unfortunately it's a LogDebug so I can't verify it in unit tests on the CI. Instead I check that the relevant mocks were invoked.